### PR TITLE
Added RPC

### PR DESCRIPTION
**Pull request by OrangutanGaming**

### CODE REVIEW

File: discord/rpc [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    """
+    def __init__(self, **options):
+        if sys.platform == 'linux':
+            env_vars = ['XDG_RUNTIME_DIR', 'TMPDIR', 'TMP', 'TEMP']
+            path = next((os.environ.get(path, None) for path in env_vars if path in os.environ), '/tmp')
+            self.ipc_path = f'{path}/discord-ipc-0'
+            self.loop = asyncio.get_event_loop()
+        elif sys.platform == 'win32':
+            self.ipc_path = r'\\?\pipe\discord-ipc-0'
+            self.loop = asyncio.ProactorEventLoop()
```
Comment (SunDwarf) [inline discord/rpc]: No. This makes a brand new loop.

File: discord/rpc [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+        self.sock_reader: asyncio.StreamReader = None
```
Comment (SunDwarf) [inline discord/rpc]: Not 3.4 compatible.

File: discord/rpc [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        self.sock_writer: asyncio.StreamWriter = None
+
+        self.client_id = options.get('client_id')
+
+    async def read_output(self):
```
Comment (SunDwarf) [inline discord/rpc]: Not 3.4 compatible.

File: discord/rpc [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+    Parameters
+    ----------
+    client_id : [str]
+        The client id of the application.
+    """
+    def __init__(self, **options):
+        if sys.platform == 'linux':
+            env_vars = ['XDG_RUNTIME_DIR', 'TMPDIR', 'TMP', 'TEMP']
+            path = next((os.environ.get(path, None) for path in env_vars if path in os.environ), '/tmp')
+            self.ipc_path = f'{path}/discord-ipc-0'
```
Comment (SunDwarf) [inline discord/rpc]: Not 3.4 compatible.

File: discord/rpc [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        self.client_id = options.get('client_id')
+
+    async def read_output(self):
+        """|coro|
+        """
+        while True:
+            data = await self.sock_reader.read(1024)
+            if data == b'':
+                self.sock_writer.close()
+                exit(0)
```
Comment (SunDwarf) [inline discord/rpc]: Why are you exiting???

File: discord/rpc [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+        await self.handshake()
+        self.send_rich_presence()
+        await self.read_output()
+
+    def close(self):
+        """|coro|
+        """
+        self.sock_writer.close()
+        self.loop.close()
+        exit(0)
```
Comment (SunDwarf) [inline discord/rpc]: See above point about exiting.

**Comment by NCPlayz** [comment]
This is a blank file. Please change to the `*.py` format.

**Comment by SunDwarf** [comment]
The lib is for 3.4. This code is for 3.6

The file isn't even named properly, it doesn't have any extension.

This is clearly taken from a standalone app and added extra comments to it to fit with the rest of lib. Try again.

**Comment by kyb3r** [comment]
This was copied from https://github.com/GiovanniMCMXCIX/PyDiscordRPC 
smh

**Pull request closed without merging**